src/plugins/intel_gpu/src/runtime/engine.cpp

```diff
@@ -254,7 +254,21 @@ std::shared_ptr<cldnn::engine> engine::create(engine_types engine_type,
                                               const engine_configuration& configuration,
                                               const InferenceEngine::ITaskExecutor::Ptr task_executor) {
     device_query query(engine_type, runtime_type);
-    device::ptr default_device = query.get_available_devices().begin()->second;
+#ifdef GPU_DEBUG_CONFIG
+#if GPU_DEBUG_CONFIG
```

Comment:
I don't really understand why single `ifdef` is not enough.
```cpp
size_t default_device_idx = 0;

#ifdef GPU_DEBUG_CONFIG
    GPU_DEBUG_GET_INSTANCE(debug_config);
    default_device_idx = debug_config->device_index;
#endif
device::ptr default_device = next(query.get_available_devices().begin(), default_device_idx)->second;
```

Reply:
I think we need to be prepared for CI test in the future, where ENABLE_DEBUG_CAPS is disabled.

That means, it should be something like this:

```
size_t default_device_idx = 0;
if (const auto env_var = std::getenv("OV_GPU_DeviceIndex")) {
    std::istringstream ss(str);
    ss >> default_device_idx;
}
device::ptr default_device = next(query.get_available_devices().begin(), default_device_idx)->second;
```

Reply:
BTW, we may consider adding helper macro for extracting something from debug config:

```cpp
#ifdef GPU_DEBUG_CONFIG
#define GPU_DEBUG_IF(cond) if (cond)
#define GET_DEBUG_CONFIG_VALUE_OR(config_field, default_value) cldnn::debug_configuration::get_instance()->config_field
#else
#define GPU_DEBUG_IF(cond) if (0)
#define GET_DEBUG_CONFIG_VALUE_OR(config_field, default_value) default_value
#endif
...

size_t default_device_idx = GET_DEBUG_CONFIG_VALUE_OR(device_index, 0);

```

Reply:
@isanghao I would want to avoid this option being exposed w/o DEBUG_CAPS. Maybe we just need to add an additional command line parameter to unit/func tests with target device id?

Reply:
I thought so. So we reviewed whether we can feed cmdline option through gtest, but could not find the way as gtest_main does not pass argv. Any better idea?

Reply:
For unit tests it's quite simple. We can just stop linking with gtest_main and add custom impl:
```cpp
int main(int argc, char **argv) {
  ::testing::InitGoogleTest(&argc, argv);
    // parse some args
   return RUN_ALL_TESTS();
}

```

Reply:
Changed link to gtest_main_gpu.cpp.
Support command line argument `-d` for device selection.